.github/workflows/android.yml

```diff
@@ -0,0 +1,29 @@
+name: Android build
+
+on:
+  push:
+    branches: [ main, '@kwasow/CI' ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 🏗 Setup repo
+        uses: actions/checkout@v3
+
+      - name: 🏗 Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: npm
+
+      - name: 📦 Install dependencies
+        run: npm install
```

Comment:
this project uses yarn, so I'd also use that in the ci action?

Reply:
Sure, but something is wrong with the repo then, because we have a `package-lock.json` which is an npm file. Yarn uses `yarn.lock`

Reply:
![image](https://github.com/user-attachments/assets/c4442b74-8dba-405a-8a33-012716156229)

The CI fails when I use yarn

Reply:
yeah I also wondered. but the package manager specified in package.json is yarn:
```
"packageManager": "yarn@4.0.2"
```

Reply:
@dansup what's the package management situation, what should we use?

Reply:
Maybe we should remove yarn then and say to use npm?

Reply:
@Simon-Laux I think there is more maintenance work to be done. I believe that `google-services.json` and the EAS project ID (in `app.json`) should both be included in the public repo. We can probably keep `google-services.json` private if @dansup really wants it, but it will be difficult to get around the missing EAS ID in a workflow.

Reply:
I think listing/formatting and typescript tests would be already be a good start for ci.
I bet also that you can somehow locally build in GitHub Actions without using expos cloud.

on deltachat project we build iOS in a GitHub action: https://github.com/deltachat/deltachat-ios/actions/runs/12766119235/workflow which works quite well, I bet android might be even easier.

Reply:
Thanks @Simon-Laux, I'll check it out. I tried building it with expo directly, skipping EAS, but I ran into weird issues when running on GitHub. Maybe even more manually is the way to go.

Reply:
I made an issue for discussing package managers #189